src/pages/Tailwind/Swap/SwapPanel.tsx

```diff
@@ -0,0 +1,388 @@
+  const [approveState, setApproveState] = useState(ApproveButtonState.NotApproved)
+  const [showModal, setShowModal] = useState(false)
+  const [showSettingsModal, setShowSettingsModal] = useState(false)
+  const [buttonState, setButtonState] = useState(ButtonState.EnterAmount)
+  const [timeLeft, setTimeLeft] = useState(60)
+  const [isExpired, setIsExpired] = useState(false)
+  const [swapTransactionModalState, setSwapTransactionModalState] = useState(ModalState.NotConfirmed)
+  const [txhash, setTxhash] = useState('')
+
+  const handleApprove = useCallback(async () => {
+    try {
+      setApproveState(ApproveButtonState.Approving)
+      const txHash: any = await approve()
+
+      // user rejected tx or didn't go thru
+      if (txHash.code === 4001 || !txHash) {
+        setApproveState(ApproveButtonState.NotApproved)
+      }
+    } catch (e) {
+      console.log(e)
+    }
+  }, [approve, setApproveState])
+
+  const startTimer = () => {
+    let maxTime = timeLeft
+    const timer = setInterval(() => {
+      maxTime -= 1
+      setTimeLeft(maxTime)
+    }, 1000)
+
+    setTimeout(() => {
+      clearInterval(timer)
+      setTimeLeft(60)
+      getMinimumAmount(fromInputValue)
+      setIsExpired(true)
+      getPrice()
+    }, 60000)
+  }
+
+  useEffect(() => {
+    getPrice()
+    getMinimumAmount(fromInputValue ? fromInputValue : '0')
+    setToInputValue(minimumAmount || '0')
+  }, [toCurrency, fromCurrency, getMinimumAmount, getPrice, fromInputValue, minimumAmount])
+
+  useEffect(() => {
+    if (chainId) {
+      setToCurrency((haloTokenList[chainId as ChainId] as Token[])[0])
+      setFromCurrency((haloTokenList[chainId as ChainId] as Token[])[1])
+    }
+  }, [chainId])
+
+  useEffect(() => {
+    if (allowance !== '' && Number(allowance) > 0) {
+      setApproveState(ApproveButtonState.Approved)
+      setButtonState(ButtonState.Swap)
+    } else {
+      setApproveState(ApproveButtonState.NotApproved)
+      setButtonState(ButtonState.EnterAmount)
+    }
+  }, [allowance])
+
+  const NotApproveContent = () => {
+    return (
+      <div className="mt-4 flex space-x-4">
+        <div className="w-1/2">
+          <ApproveButton
+            title="Approve"
+            state={ApproveButtonState.NotApproved}
+            onClick={() => {
+              if (fromInputValue && toInputValue) {
+                handleApprove()
+              }
+            }}
+          />
+        </div>
+        <div className="w-1/2">
+          <PrimaryButton title="Swap" state={PrimaryButtonState.Disabled} />
+        </div>
+      </div>
+    )
+  }
+
+  const SwapContent = () => {
+    return (
+      <div className="mt-4">
+        <PrimaryButton
+          type={PrimaryButtonType.Gradient}
+          title="Swap"
+          state={PrimaryButtonState.Enabled}
+          onClick={() => {
+            if (fromInputValue && toInputValue && Number(fromInputValue) > 0 && Number(toInputValue) > 0) {
```

Comment:
Same here.  Might be overkill?

```suggestion
            if (Number(fromInputValue) > 0 && Number(toInputValue) > 0) {
```

Reply:
changed this to a different condition already!